### PR TITLE
Attempt to fix Flipper-UI

### PR DESCRIPTION
**Pull request by joshpuetz**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Another try at fixing ongoing issue with the Flipper/Feature Flag UI (https://github.com/thepracticaldev/tech-private/issues/450). After some investigation: I found that omitting the `Origin` http header when using curl allowed me to successfully POST to production. Digging through both Flipper's forums and Molly's similar Sidekiq fix from last year, I suspect something is happening with the CSRF protection FlipperUI is getting from Sinatra.

(Related: I had to pin the `redis-store` gem at 1.9.0 to work around an issue noted here:  https://github.com/redis-store/redis-store/pull/333

## Related Tickets & Documents
https://github.com/thepracticaldev/tech-private/issues/450

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -42,7 +42,13 @@
 
     authenticate :user, ->(user) { user.has_role?(:tech_admin) } do
       mount Blazer::Engine, at: "blazer"
-      mount Flipper::UI.app(Flipper, { rack_protection: {} }), at: "feature_flags"
+
+      flipper_ui = Flipper::UI.app(Flipper) do |builder|
+        builder.use Rack::Protection, origin_whitelist: ["https://dev.to"]
```
Comment (citizen428) [inline config/routes.rb:47]: **suggestion:** We should be able to leverage our `URL` helper module here, so we don't add anything new that later needs to be generalized again.

```suggestion
        builder.use Rack::Protection, origin_whitelist: URL.url
```

File: Gemfile [hunk 2]
```diff
@@ -85,6 +85,7 @@ gem "recaptcha", "~> 5.5", require: "recaptcha/rails" # Helpers for the reCAPTCH
 gem "redcarpet", "~> 3.5" # A fast, safe and extensible Markdown to (X)HTML parser
 gem "redis", "~> 4.2.1" # Redis ruby client
 gem "redis-rails", "~> 5.0.2" # Redis for Ruby on Rails
+gem "redis-store", "~> 1.9.0" # Redis storage options for Rails (listed here to lock version above 1.9.0 - see https://github.com/redis-store/redis-store/pull/333)
```
Comment (rhymes) [inline Gemfile:88]: I don't think we need to add this to the Gemfile as it's not an explicit dependency, I ran `bundle update redis-store` on master and it did update it nonetheless.

Maybe we can move `see https://github.com/redis-store/redis-store/pull/333` and a short explanation about what's going on in the routes file

File: config/routes.rb (hunk 1 above)
Comment (joshpuetz) [reply config/routes.rb:47]: This is a great suggestion!

File: Gemfile (hunk 2 above)
Comment (joshpuetz) [reply Gemfile:88]: I swear version Redis-store 1.9 wasn't updating for me, but now that 1.9 is in `Gemfile.lock` I agree this isn't needed in the Gemfile anymore! I'll move this PR link to routes per your suggestion.

**Comment by joshpuetz** [comment]
(Note that Github is having some issues showing newer commits to PR, you can see my latest here: https://github.com/forem/forem/compare/master...joshpuetz/flipper-ui-session-fix)

**Comment by joshpuetz** [comment]
Update: it didn't work 😧 Info added to issue https://github.com/thepracticaldev/tech-private/issues/450

**Pull request merged**